app/Models/Signup.php

```diff
@@ -65,21 +65,62 @@ public function posts()
     }
 
     /**
-     * Get the visible posts associated with this signup.
+     * Get the visible posts associated with this signup, optionally by type.
      *
+     * @param string Optional param to specify post type.
      * @return \Illuminate\Database\Eloquent\Relations\HasMany
      */
-    public function visiblePosts()
+    public function visiblePosts($type = null)
     {
         if (! is_staff_user()) {
-            return $this->hasMany(Post::class)->where('status', '=', 'accepted')
-                                              ->orWhere('northstar_id', auth()->id())
-                                              ->with('tags');
+            if ($type === null) {
+                return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+                                                  ->orWhere('northstar_id', auth()->id())
+                                                  ->with('tags');
+            }
+
+            if (count($type) > 1) {
+                return $this->hasMany(Post::class)->whereIn('type', array_values($type))
+                                                  ->where('status', '=', 'accepted')
+                                                  ->orWhere('northstar_id', auth()->id())
+                                                  ->with('tags')
+                                                  ->get();
+                // return $this->hasMany(Post::class)->where(function ($q) use ($type) {
+                //     $q->whereIn('type', array_values($type))
+                //       ->where('status', '=', 'accepted')
+                //       ->orWhere('northstar_id', auth()->id())
+                //       ->with('tags')
+                //       ->get();
+                // });
+
```

Comment:
Everything is working except when using include params as an owner (e.g. `api/v3/signups?include=posts:type(text|photo)` still returns `voter-reg` posts. I tried to do what is commented out but didn't return any posts. I'm wondering if there is a certain order I'd have to use `whereIn`, `where`, and `orWhere` when using them all together?

Reply:
@chloealee What query is this making? I'm thinking you might need to use [parameter grouping](https://laravel.com/docs/5.6/queries#parameter-grouping) so that these conditions get applied with parenthesis in the right place...

Reply:
@DFurnes I tried to do that (see things I've tried commented out!) but didn't work. Maybe I'm doing it incorrectly?